menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -115,18 +121,18 @@ class SchemaController @Autowired() (
 
         val sparkStruct = StructType(sparkMenasConvertor.convertMenasToSparkFields(schema.fields))
         if (pretty) sparkStruct.prettyJson else sparkStruct.json
-      case None         =>
+      case None =>
         throw notFound()
     }
   }
 
   /**
-    * Parses an StructType JSON file contents and converts it to Spark [[StructType]].
-    *
-    * @param structTypeJson A StructType JSON string.
-    * @return The parsed schema as an instance of [[StructType]].
-    */
-  private def parseStructType(structTypeJson: String): StructType = {
+   * Parses an StructType JSON file contents and converts it to Spark [[StructType]].
+   *
+   * @param structTypeJson A StructType JSON string.
+   * @return The parsed schema as an instance of [[StructType]].
+   */
+  private[controllers] def parseStructType(structTypeJson: String): StructType = {
```

Comment:
I am not fond of opening `private` methods just for the sake of testing.
I would prefer testing or higher level. Or even using reflection to get the private method. We can discuss it on CQC (in chat).

Reply:
Yeah, let's discuss this on CQC (meeting or channel). And if we decide this is an issue, we can fix this in the next PR.
My personal take on this - I'm fine with the way it is.

Reply:
It is tested on a higher level as well, e.g. at `SchemaApiIntegrationSuite.scala:722`

Reply:
Well then if also the failure tests would be moved to higher level, the lower ones wouldn't be needed. (Too many tests testing the same test cost upkeep efforts and build time.)
Also IMHO the opening is usually a code smell, in this case that `SchemaController` breaks the S from SOLID.

Reply:
Ok.
How about moving `parse*` method to a separate object, say, `SchemaParsers`, make all these methods `public` and leave tests as is?

The higher-level test here is an integration test. The lower level ones are normal unit tests. Let's leave both, they serve their purpose.

Reply:
I think @yruslan's suggestion to move them outside the Controller makes perfect sense. No need to open up the access modifiers, and lightens the load of what the Controller is meant to do - provide piping for services which hold business logic without holding any itself.

Reply:
> Ok.
> How about moving `parse*` method to a separate object, say, `SchemaParsers` ...

I was actually contemplating something like this before, but did not want to cause too _ado  about nothing_ 😄

Cool, I will try to introduce a separate unit-testable object for it.

Reply:
Done as suggested (more or less), hope you guys like it.